File: app/src/main/java/com/bove/martin/pexel/data/retrofit/PexelService.java
package com.bove.martin.pexel.data.retrofit;

import com.bove.martin.pexel.data.model.Foto;
import com.bove.martin.pexel.utils.ApiKey;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;

/**
 * Created by Martín Bove on 28/05/2018.
 * E-mail: mbove77@gmail.com
 */
public interface PexelService {
    @Headers("Authorization: " + ApiKey.PEXELS_TOKEN)
    @GET("curated")
    Call<List<Foto>> getCurated(@Query("per_page") int itemNum, @Query("page") int numPage);

    @Headers("Authorization: " + ApiKey.PEXELS_TOKEN)
    @GET("search")
    Call<List<Foto>> getSearch(@Query("query") String query, @Query("per_page") int itemNum, @Query("page") int numPage);

    @Headers("Authorization: " + ApiKey.PEXELS_TOKEN)
    @GET("photos")
    Call<List<Foto>> getPhoto(@Query("id") String id);
}

File: app/src/main/java/com/bove/martin/pexel/data/retrofit/PexelService.kt
package com.bove.martin.pexel.data.retrofit

import com.bove.martin.pexel.data.model.Foto
import com.bove.martin.pexel.utils.ApiKey
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query

/**
 * Created by Martín Bove on 28/05/2018.
 * E-mail: mbove77@gmail.com
 */
interface PexelService {
    @Headers("Authorization: " + ApiKey.PEXELS_TOKEN)
    @GET("curated")
    suspend fun getCurated(@Query("per_page") itemNum: Int, @Query("page") numPage: Int): Response<List<Foto>>

    @Headers("Authorization: " + ApiKey.PEXELS_TOKEN)
    @GET("search")
    suspend  fun getSearch(@Query("query") query: String?, @Query("per_page") itemNum: Int, @Query("page") numPage: Int): Response<List<Foto>>

    @Headers("Authorization: " + ApiKey.PEXELS_TOKEN)
    @GET("photos")
    suspend fun getPhoto(@Query("id") id: String?): Response<List<Foto>>
}